src/app/filters/filter-fields.component.ts

```diff
@@ -27,11 +30,16 @@ export class FilterFieldsComponent implements OnInit {
   @Input() config: FilterConfig;
 
   @Output('onAdd') onAdd = new EventEmitter();
+  @Output('onSelectType') onSelecttype = new EventEmitter();
+
+  @ViewChild('searchField') searchField: any;
+  @ViewChild('listData') listData: any;
```

Comment:
It supports one particular section in the DOM but that doesn't mean it supports one field only. That section in DOM is used for all the fields. I tested with multiple typeahead field and it's working fine with that. In the example itself, if you try replacing all **select** s to **typeahead** s it works fine.